### PR TITLE
FiBo-197 backend customise abstract user

### PR DESCRIPTION
**Pull request by ChrisEwert**
We changed the AbstractUser. It no longer uses username, first_name, last_name and is_staff. 
We removed every occurrence of a username and it instead uses email for everything. We changed the tests and the documentation too.

### CODE REVIEW

File: docs/use_cases/UC_0_incomes_expenses/expenses_income.drawio.svg
Comment (Jonas-Karl) [inline docs/use_cases/UC_0_incomes_expenses/expenses_income.drawio.svg:1]: What did you change here?

File: docs/use_cases/UC_3_change_added_data/change_added_data.drawio.svg
Comment (Jonas-Karl) [inline docs/use_cases/UC_3_change_added_data/change_added_data.drawio.svg:1]: What did you change in this file?

File: backend/api/models.py
Comment (Jonas-Karl) [inline backend/api/models.py:1]: in this file, there are some untested lines of codes which. Could you add the according tests?

**Review by Jonas-Karl — CHANGES_REQUESTED** [review]
Looks good to me. I think the only thing that needs to be added are some tests for edge cases :)

**Comment by Jonas-Karl** [comment]
Does anybody know why the Android tests fail in this branch?

File: backend/api/models.py
Comment (ChrisEwert) [reply backend/api/models.py:1]: I have no idea, how to do that :D

File: docs/use_cases/UC_0_incomes_expenses/expenses_income.drawio.svg
Comment (ChrisEwert) [reply docs/use_cases/UC_0_incomes_expenses/expenses_income.drawio.svg:1]: I removed every instance of "username". Even in the drawio.svg files. I guess somehow there was a username before? I dont know where

File: .vscode/settings.json
Comment (XSnelliusX) [inline .vscode/settings.json:1]: This should not be on Git can you please add it to the gitignore?

**Review by Jonas-Karl — CHANGES_REQUESTED** [review]

**Review by Jonas-Karl — APPROVED** [review]

File: .vscode/settings.json
Comment (Jonas-Karl) [inline .vscode/settings.json:1]: this file can be deleted / you can put it into gitignore

**Review by jens1o — APPROVED** [review]

**Pull request merged**